fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/PageLinkBuilder.java

```diff
@@ -0,0 +1,94 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web.apirequest;
+
+import com.yahoo.bard.webservice.util.Pagination;
+import com.yahoo.bard.webservice.web.util.PaginationLink;
+
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Link;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+import javax.ws.rs.core.UriInfo;
+
+public class PageLinkBuilder {
```

Comment:
Would be good to get a doc on the so it's clear what it's responsibilities are.

Reply:
Good call.

I think I phoned it in on this class because I don't think it's a long term parking spot for these responsibilities.  Eventually I want to make the building of response headers more injectable, but I had enough change in flight I didn't want to add a lot more.